src/standalone/activation/globalActivation.ts

```diff
@@ -61,6 +67,11 @@ export class GlobalActivation implements IExtensionSingleActivationService {
                 vscode.languages.registerCodeLensProvider([PYTHON_FILE_ANY_SCHEME], this.dataScienceCodeLensProvider)
             );
         }
+        if (this.pythonCellFoldingProvider) {
+            this.extensionContext.subscriptions.push(
```

Comment:
Instead of modiying globalActivation you can do this in the `activate` method of the folding provider.

Reply:
YOu might as well register these providers in the activate method of the `FoldingProvider` class itself.

Have a look at the `NotebookCellBangInstallDiagnosticsProvider` & `PythonKernelCompletionProviderRegistration`
Thats what we do in other places as well, else we unnecessarily inject these dependencies into the `globalActivation.ts` file or other places.

Reply:
thanks for pointing that out - I was following the codeLensProvider, so that should probably also be moved: https://github.com/microsoft/vscode-jupyter/issues/11424